packages/rich-text/src/plugins/CommandPalette/CommandPaletteService.js

```diff
@@ -86,40 +86,45 @@ export const isEmbeddingEnabled = field =>
   isNodeTypeEnabled(field, INLINES.EMBEDDED_ENTRY);
 
 export class CommandPaletteActionBuilder {
-  constructor(field, permissions) {
-    this.field = field;
-    this.permissions = permissions;
+  constructor(sdk) {
+    this.sdk = sdk;
   }
 
   // TODO: Let's create dedicated functions for assets so we do not have to pass a CT.
 
   maybeBuildEmbedAction(embedType, contentType, callback) {
-    if (!isNodeTypeEnabled(this.field, embedType)) {
+    if (!isNodeTypeEnabled(this.sdk.field, embedType)) {
       return false;
     }
     const isAsset = !contentType;
-    if (!isAsset && !isValidLinkedContentType(this.field, contentType, embedType)) {
+    if (!isAsset && !isValidLinkedContentType(this.sdk.field, contentType, embedType)) {
       return false;
     }
 
     return buildAction(embedType, contentType, ACTIONS.EMBED, callback);
   }
 
-  maybeBuildCreateAndEmbedAction(embedType, contentType, callback) {
-    if (!isNodeTypeEnabled(this.field, embedType)) {
+  async maybeBuildCreateAndEmbedAction(embedType, contentType, callback) {
+    if (!isNodeTypeEnabled(this.sdk.field, embedType)) {
       return false;
     }
 
     const isAsset = !contentType;
     if (!isAsset) {
-      if (!isValidLinkedContentType(this.field, contentType, embedType)) {
+      if (!isValidLinkedContentType(this.sdk.field, contentType, embedType)) {
         return false;
       }
-      if (!this.permissions.canCreateEntryOfContentType(contentType.sys.id)) {
+
+      const canCreateContentType = await this.sdk.access.can('create', contentType);
```

Comment:
:exclamation: Wait... whether or not you can create a CT vs. whether or not you can create an entry of a certain CT is something completely different. Is it just a naming issue here and this should be named `canCreateEntryOfCT` or is your intention to call `this.sdk.access.can('create', 'ContentType')`?

I guess it's a naming issue but then I am confused by `await this.sdk.access.can('create', contentType);`. I am looking [at the documentation](https://www.contentful.com/developers/docs/extensibility/ui-extensions/sdk-reference/) and I must say it is pretty confusing for this case. I believe `can` does too many things in one single function, or maybe it can't do what you're trying to do here?

E.g. `can('create', 'Entry', contentType)` would make more sense to me, otherwise, look at `can('update', contentType)` vs. `can('create', contentType)` where it's pretty ambiguous (and not documented) what either of these would actually verify exactly :thinking:

Reply:
Good catch! It was incorrect indeed. Now it should be fixed.